### PR TITLE
added more information to the decoding exceptions

### PR DESCRIPTION
**Pull request by vpodlesnyak**
added the mapper name and the field name to the exception message

### CODE REVIEW

**Comment by thoutbeckers** [comment]
has a failed JS test but I assume it's not new?

**Review by avdyushin — APPROVED** [review]

File: firebase-common/src/commonMain/kotlin/dev/gitlive/firebase/decoders.kt [hunk 1]
```diff
@@ -122,18 +131,35 @@ open class FirebaseCompositeDecoder constructor(
...
 
     @ExperimentalSerializationApi
     override fun decodeInlineElement(descriptor: SerialDescriptor, index: Int): Decoder =
-        FirebaseDecoder(get(descriptor, index))
+        decodeElement(descriptor, index, ::FirebaseDecoder)
+
+    private fun <T> decodeElement(descriptor: SerialDescriptor, index: Int, decoder: (Any?) -> T): T {
+        return try {
+            decoder(get(descriptor, index))
+        } catch (e: Exception) {
```
Comment (leokapanen) [inline firebase-common/src/commonMain/kotlin/dev/gitlive/firebase/decoders.kt:156]: `} catch (exception: Exception) {` or `} catch (error: Exception) {` will be better

**Review by leokapanen — APPROVED** [review]

**Comment by leokapanen** [comment]
Please fix a test `> Task :firebase-firestore:jsLegacyTest FAILED`

File: firebase-common/src/commonMain/kotlin/dev/gitlive/firebase/decoders.kt (hunk 1 above)
Comment (avdyushin) [reply firebase-common/src/commonMain/kotlin/dev/gitlive/firebase/decoders.kt:156]: +1 for `exception`

**Comment by vpodlesnyak** [comment]
>> has a failed JS test but I assume it's not new?
the tests are sometimes failing for js, it's ok after rerun. I didn't have time to investigate yet

**Pull request merged**